src/main/java/lesson19/template/OfficesDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package lesson19.template;
+
+import lesson17.OJDBCUtils;
+import lesson17.models.Offices;
+import lesson17.models.Orders;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import static lesson17.OJDBCUtils.setOfficeValuesToStatement;
+
+public class OfficesDaoImpl {
+
+    public static class OfficeInsert extends abstractDao {
+        @Override
+        public <K> PreparedStatement updateTable(K tableObj, Connection conn) throws SQLException {
+            Offices office = (Offices) tableObj;
```

Comment:
Why do you do type conversions if you can just pass the office to the method?

Reply:
realy?

Reply:
Actualy to use generics, it drain off almost the half of code. We can use generics in the class scope, but but it would not cancel the cast

Reply:
But first idea was to write one method for diferent classes